File: Python/Sorting-0-1-2_without_algo.py
class Solution:
    def sort012(self, arr, n):
        # counts all individual numbers in list using count function
        count0, count1, count2 = arr.count(0), arr.count(1), arr.count(2)

        # list slicing is done to insert desired numbers in designated positions
        arr[0:count0] = [0] * count0
        arr[count0:count0 + count1] = [1] * count1
        arr[count0 + count1:] = [2] * count2



if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = [int(x) for x in input().strip().split()]
        ob = Solution()
        ob.sort012(arr, n)
        for i in arr:
            print(i, end=' ')
        print()

File: C++/Sorting-0-1-2_without_algo.cpp
#include<bits/stdc++.h>
using namespace std;


class Solution
{
public:
    void sort012(int a[], int n)
    {
        //counting all the 0's , 1's and 2's
        int count0 = 0, count1 = 0, count2 = 0;

        for (int i = 0;i < n;++i)
            if (a[i] == 0)
                ++count0;
            else if (a[i] == 1)
                ++count1;
            else
                ++count2;

        //updating the array with total 0's , 1's and 2's
        for (int i = 0;i < n;++i)
            if (count0 > 0)
            {
                a[i] = 0;
                --count0;
            }
            else if (count1 > 0)
            {
                a[i] == 1;
                --count1;
            }
            else
                a[i] = 2;
    }

};


int main()
{
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        int a[n];
        for (int i = 0;i < n;i++) {
            cin >> a[i];
        }

        Solution ob;
        ob.sort012(a, n);

        for (int i = 0;i < n;i++) {
            cout << a[i] << " ";
        }

        cout << endl;


    }
    return 0;
}
